pkgs/games/angband/default.nix

```diff
@@ -27,11 +43,24 @@ stdenv.mkDerivation rec {
   configureFlags = lib.optional enableSdl2 "--enable-sdl2";
 
   installFlags = [ "bindir=$(out)/bin" ];
+}
+else stdenv.mkDerivation {
```

Comment:
I don't think you need two `mkDerivations` for darwin support to happen, especially if everything is built from source. Try using

- `buildInputs = lib.optionals stdenv.isLinux [ ... ] ++ lib.optionals stdenv.isDarwin [ ... ]`
- `buildPhase = if stdenv.isDarwin then "something" else null`
- `makefile = if stdenv.isDarwin then "Makefile.osx" else "Makefile"`

to fit the logic into one.